app/gui/src/dashboard/providers/BindingFocusScopeProvider.tsx

```diff
@@ -0,0 +1,11 @@
+/** @file Provider for focus scope for input bindings. */
+import { createContext, useContext } from 'react'
+
+export const BindingFocusScopeContext = createContext<React.RefObject<HTMLElement>>({
+  current: document.body,
+})
+
+/** Get the binding focus scope from its context. */
+export function useBindingFocusScope() {
+  return useContext(BindingFocusScopeContext)
+}
```

Comment:
Why do we create a context which is never changed?

In its only usage we even apply `?? document.body` anyway.